### PR TITLE
Refactor/alw integration

### PR DESCRIPTION
**Pull request by martind260**
---
name: Pull request
about: Contribute your changes to the project
title: '[pull-request]'
labels: ''
assignees: ''
---
**Description**
Short description or comments

**Reference**
Issues https://app.zenhub.com/workspaces/digiwf-621f70bf50ea1100120b7e93/issues/it-at-m/digiwf-alw-integration/11

**Checklist**

- [x] Acceptance criteria are met
- [x] Pipeline has been run successfully
- [ ] JUnit tests have been run successfully
- [x] Changelog is adjusted
- [ ] [Documentations](https://git.muenchen.de/groups/digitalisierung/-/wikis/Dokumentationen) are completed
- [ ] Frontend is tested
- [ ] Created sub-branches are deleted
- [x] Boards are updated ( [Digitalisierung](https://git.muenchen.de/groups/digitalisierung/-/boards) , [Support](https://git.muenchen.de/digitalisierung/digiwf-support) , [Zenhub](https://app.zenhub.com/workspaces/digiwf-621f70bf50ea1100120b7e93/board) )
- [ ] feature process is created or extended
- [ ] [git-ops](https://git.muenchen.de/digitalisierung/digiwf-ops) is customized
- [ ] Openshift environments are prepared (Secrets, etc.)

### CODE REVIEW

File: digiwf-gateway/src/main/resources/application-local.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ server:
 spring:
 
   # Define the local keycloak realm here
-  realm: <YOUR_LOCAL_REALM>
+  realm: P82
```
Comment (lmoesle) [inline digiwf-gateway/src/main/resources/application-local.yml:6]: Overwrite this setting in an env variable or in application-internal.properties?

File: digiwf-integrations/digiwf-alw-integration/README.md [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+
+  <p align="center">
+    Spring-Boot-Starter project to integrate the ALW-System into DigiWF
+    <br /><a href="https://github.com/it-at-m/digiwf-alw-integration/issues">Report Bug</a>
+    ·
+    <a href="https://github.com/it-at-m/digiwf-alw-integration/issues">Request Feature</a>
+  </p>
+</div>
+
+<!-- ABOUT THE PROJECT -->
```
Comment (lmoesle) [inline digiwf-integrations/digiwf-alw-integration/README.md]:
```suggestion

```
I would remove the header section from all README.mds excepting the root one.

File: digiwf-integrations/digiwf-alw-integration/README.md [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+
+The goal of this library is to enable async communication with the ALW System dispatched by an EventBus of your environment.
+
+Features:
+
+* Can be used to dispatch requests/responses of the ALW Personeninfo Feature asynchronously through an eventbus.
+* Can inform the receiver through an eventbus if the request was successful or if there was a problem.
+* Performs a functional ping to the ALW System to check connectivity.
+
+<p align="right">(<a href="#top">back to top</a>)</p>
```
Comment (lmoesle) [inline digiwf-integrations/digiwf-alw-integration/README.md]:
```suggestion

```

File: digiwf-integrations/digiwf-alw-integration/README.md [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+
+<!-- ROADMAP -->
+## Roadmap
+
+See the [open issues](#) for a full list of proposed features (and known issues).
+
+<p align="right">(<a href="#top">back to top</a>)</p>
+
```
Comment (lmoesle) [inline digiwf-integrations/digiwf-alw-integration/README.md]:
```suggestion

```

File: digiwf-integrations/digiwf-alw-integration/README.md [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+
+For an example, please refer to the [example project](https://github.com/it-at-m/digiwf-alw-integration/tree/dev/example-digiwf-alw-integration).
+There you can:
+* Configure the example application (see above)
+* Start the example application
+* Make a http request to the configured test endpoints from <i>[ExampleController](https://github.com/it-at-m/digiwf-alw-integration/blob/dev/example-digiwf-alw-integration/src/main/java/io/muenchendigital/digiwf/alw/integration/api/controller/ExampleController.java) </i> on http://localhost:10006/testGetAlwZustaendigkeitEventBus
+* Observe the output in the console
+
+<p align="right">(<a href="#top">back to top</a>)</p>
```
Comment (lmoesle) [inline digiwf-integrations/digiwf-alw-integration/README.md]:
```suggestion

```

File: digiwf-integrations/digiwf-alw-integration/README.md [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+
+[stars-url]: https://github.com/it-at-m/digiwf-alw-integration/stargazers
+
+[issues-shield]: https://img.shields.io/github/issues/it-at-m/digiwf-alw-integration.svg?style=for-the-badge
+
+[issues-url]: https://github.com/it-at-m/digiwf-alw-integration/issues
+
+[license-shield]: https://img.shields.io/github/license/it-at-m/digiwf-alw-integration.svg?style=for-the-badge
+
+[license-url]: https://github.com/it-at-m/digiwf-alw-integration/blob/master/LICENSE
```
Comment (lmoesle) [inline digiwf-integrations/digiwf-alw-integration/README.md]:
```suggestion

```

File: digiwf-integrations/digiwf-alw-integration/digiwf-alw-integration-core/pom.xml [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+        <dependency>
+            <groupId>org.springframework</groupId>
+            <artifactId>spring-web</artifactId>
+        </dependency>
+
+        <!-- streaming -->
+        <dependency>
+            <groupId>io.muenchendigital.digiwf</groupId>
+            <artifactId>digiwf-spring-cloudstream-utils</artifactId>
+            <version>${digiwf-spring-cloudstream-utils.version}</version>
```
Comment (lmoesle) [inline digiwf-integrations/digiwf-alw-integration/digiwf-alw-integration-core/pom.xml]:
```suggestion
            <version>${project.version}</version>
```
Use the head version (project version) of the spring-cloudstream-utils. The spring-cloudstream-utils are already included in the monorepo.

File: digiwf-integrations/digiwf-alw-integration/digiwf-alw-integration-starter/pom.xml [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter</artifactId>
+            <version>${spring.boot.version}</version>
+        </dependency>
+
+        <!-- streaming -->
+        <dependency>
+            <groupId>io.muenchendigital.digiwf</groupId>
+            <artifactId>digiwf-spring-cloudstream-utils-starter</artifactId>
+            <version>${digiwf-spring-cloudstream-utils.version}</version>
```
Comment (lmoesle) [inline digiwf-integrations/digiwf-alw-integration/digiwf-alw-integration-starter/pom.xml]:
```suggestion
            <version>${project.version}</version>
```

File: digiwf-integrations/digiwf-alw-integration/pom.xml [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+    </parent>
+
+    <artifactId>digiwf-alw-integration</artifactId>
+    <name>digiwf-alw-integration</name>
+    <packaging>pom</packaging>
+    <description>Integration of the ALW-System used by DigiWF</description>
+
+    <properties>
+        <digiwf-spring-cloudstream-utils.version>0.0.1</digiwf-spring-cloudstream-utils.version>
+    </properties>
```
Comment (lmoesle) [inline digiwf-integrations/digiwf-alw-integration/pom.xml]:
```suggestion

```
The spring-cloudstream-utils are already included in the monorepo. No need to define the version here

**Review by lmoesle — APPROVED** [review]
👍🏻 well done. I just found a few minor things.

File: digiwf-gateway/src/main/resources/application-local.yml (hunk 1 above)
Comment (martind260) [reply digiwf-gateway/src/main/resources/application-local.yml:6]: If you think this is kind of a secret I can revert that. But i dont think so.

Comment (martind260) [reply digiwf-gateway/src/main/resources/application-local.yml:6]: @lmoesle What do you think?

Comment (lmoesle) [reply digiwf-gateway/src/main/resources/application-local.yml:6]: Works for me

**Pull request merged**